### PR TITLE
Add CamelCase snake_case motivation

### LINKED ISSUE #401: Snake case and Camel case: add some background explanation and logic
The goal after the cleanup is that classes are using Camel case and parameters and slots are using Snake case. We should add some explanation and the logic to the documentation so that new and old users understand the logic and old users are aware that their code may need change in parameter names to run with the new version.

To do:
- [ ] Add paragraph on this to the relevant part in https://github.com/Roche/crmPack/blob/main/vignettes/migration_from_the_old_crmPack.Rmd

### PR DESCRIPTION
**Pull request by 0liver0815**
Closes #401

### CODE REVIEW

File: vignettes/migration_from_the_old_crmPack.Rmd [hunk 1]
```diff
@@ -46,10 +51,10 @@ class_changes <- tribble(
   "DALogisticLogNormal", "", "C_par", "c_par", FALSE, "",
   "DALogisticLogNormal", "", "conditionalPEM", "cond_pem", FALSE, "",
   "TITELogisticLogNormal", "", "weightMethod", "weight_method", FALSE, "",
-  "OneParLogNormalPrior", "", "skeletonFun", "skel_fun", FALSE, "",
+  "OneParLogNormalPrior", "", "", "skel_fun", FALSE, "",
```
Comment (danielinteractive) [inline vignettes/migration_from_the_old_crmPack.Rmd:54]: are these other changes here intended?

**Review by danielinteractive — COMMENTED** [review]
thanks @0liver0815 , looks good, just double checking the table changes

**Comment by 0liver0815** [comment]
When writing the explanation a came across the following:

- Unfortunately we are not consisting with the naming. A new `class` `positive_number` is introduced, which do not follow the `CamelCase` convention for `class`.
- We do not follow strictly follow `UpperCamelCase`, but also not `lowerCamelCase`, so that `method` names like `biomarker` do not fit. However, this is probably neglectable.
- The content of the vignette do not only cover changes from 1.0.3 to the new version, but also new `classes` as `OneParLogNormalPrior` or `TITELogisticLogNormal`. Probably we need to split the vignette or at least have separate sections for the migration and new classes.

@danielinteractive please let me know if you agree. I can create an issue and work on separating new `classes` and `methods` from changed ones.

File: vignettes/migration_from_the_old_crmPack.Rmd (hunk 1 above)
Comment (0liver0815) [reply vignettes/migration_from_the_old_crmPack.Rmd:54]: Yes, please see my comment below. `OneParLogNormalPrior` is new, e.g. not part of 1.0.3. However we changed the name during development, but that is nothing we need to track (in my opinion).

**Comment by danielinteractive** [comment]
> When writing the explanation a came across the following:
> 
> * Unfortunately we are not consisting with the naming. A new `class` `positive_number` is introduced, which do not follow the `CamelCase` convention for `class`.

Hm right. Maybe add an "Exceptions" paragraph or so and explain that here we want the name close to "numeric" etc.

> * We do not follow strictly follow `UpperCamelCase`, but also not `lowerCamelCase`, so that `method` names like `biomarker` do not fit. However, this is probably neglectable.

Yeah can neglect or add to exceptions. 

> * The content of the vignette do not only cover changes from 1.0.3 to the new version, but also new `classes` as `OneParLogNormalPrior` or `TITELogisticLogNormal`. Probably we need to split the vignette or at least have separate sections for the migration and new classes.

OK good to split into separate tables, would keep in the same vignette though
> 
> @danielinteractive please let me know if you agree. I can create an issue and work on separating new `classes` and `methods` from changed ones.

maybe can do here, since this PR cleans up the migration vignette a bit more anyway

**Comment by 0liver0815** [comment]
Will do. I will let you when a review makes sense.

File: vignettes/migration_from_the_old_crmPack.Rmd (hunk 1 above)
Comment (danielinteractive) [reply vignettes/migration_from_the_old_crmPack.Rmd:54]: correct, agree

**Comment by 0liver0815** [comment]
@danielinteractive I think you can review now. I have rewritten the first section from the user perspective, i.e. the user needs to know about new classes, renamed classes and changed slots, but not necessarily about changes in methods. If I am not mistaken, a user may not use methods directly so, changes in parameter names are not important. I moved the section to the end of the vignette. In addition generic classes will not directly be used by the user and need to be be mentioned for migration. I moved that table also to the end and removed any class that were not part of v1.03 or mentioned already at the beginning of the vignette.

File: vignettes/migration_from_the_old_crmPack.Rmd [hunk 2]
```diff
@@ -21,7 +22,283 @@ The following vignette describes the changes that were introduced to the `crmPac
...
+
+class_name <- "LogisticNormal"
+eval(parse(text = paste0(".Default", class_name, "()")))
+```
+
+# Handling of `NA` or placebo returned as next dose
+
+For consistent handling how the study is stopped and to facilitate analysis of stop reasons in the operation characteristics, the handling of `NA` and placebo returned by `nextBest` methods is changed. In the previous version of `crmPack` stopping for placebo or `NA` returned by a `nextBest` method was handled automatically in the generic `Stopping` method. This is now moved into a new stopping rule `StoppingMissingDose`. As a consequence, the stopping rule `StoppingMissingDose` must be specified for those `nextBest` methods that can return NA, or when placebo is used. Otherwise the simulation may run into an error if the study is not stopped when  `NA` is returned as the next dose. `nextBest` methods that can return `NA` are `NextBestNCRM`, `NextBestNCRMLoss` and `NextBestDualEndpoint`.
+
+# Evaluation of stopping rules at a specific dose
```
Comment (danielinteractive) [inline vignettes/migration_from_the_old_crmPack.Rmd:296]: Can you please double check the levels of the section headers? e.g. this I don't think it should be top level? (but `##` or `###` even)

**Review by danielinteractive — COMMENTED** [review]
Thanks @0liver0815 for the updates! Sounds great. Just one question on the section hierarchy

**Review by danielinteractive — APPROVED** [review]

**Pull request merged**